File: 6. Zigzag Conversion/6.py
class Solution:
    def convert(self, s: str, numRows: int) -> str:
        rows = [""]*numRows
        growing = True
        counter = 0
        zigzag = ""

        if numRows == 1:
            return s

        for i in s:
            rows[counter] += i
            if growing:

                if counter == numRows -1:
                    growing = False
                    counter -= 1
                    continue

                counter += 1

            else:
                if counter == 0:
                    growing = True
                    counter += 1
                    continue
                
                counter -= 1

        for i in rows:
            zigzag += i
        
        return zigzag
    

print(Solution().convert("ABC", 1))

File: 6. Zigzag Conversion/6.cs
using System.Diagnostics;

public class Solution {
    public string Convert(string s, int numRows) {
        bool growing = true;
        int counter = 0;
        string zigzag = "";
        string[] rows = new string[numRows];

        if (numRows == 1)
        {
            return s;
        }

        foreach (char c in s)
        {
            if (growing)
            {
                rows[counter] = rows[counter] + c;

                if (counter == numRows - 1)
                {
                    growing = false;
                    counter--;
                    continue;
                }
                counter++;
            }
            else
            {
                rows[counter] = rows[counter] + c;

                if (counter == 0)
                {
                    growing = true;
                    counter++;
                    continue;
                }
                counter--;
            }
        }

        foreach (string str in rows)
        {
            zigzag = zigzag + str;
        }

        return zigzag;
    }
}